src/value.h

```diff
@@ -368,6 +368,13 @@ RSValue *RS_Int64Val(int64_t ii);
  */
 RSValue *RSValue_NewArray(RSValue **vals, uint32_t len);
 
+/**
+ * Helper function to allocate memory before passing it to RSValue_NewArray
+ */
+static inline RSValue **RSValue_AllocateArray(uint32_t len) {
+  return (RSValue **)rm_malloc(len * sizeof(RSValue *));
```

Comment:
calloc :)

Reply:
Don’t we care about the performance difference? In all the places we fill the entire array anyway. This is not a partially initialized struct but an array of pointers.
I added a `memset` call in the only place we might not fill the array (and in this case it also most likely a but in the trieMap or its iterator